tests/test_meter_llm.py

```diff
@@ -0,0 +1,320 @@
+            self.output_tokens = output_tokens
+
+    class Meta:
+        def __init__(self, billed_units):
+            self.billed_units = billed_units
+
+    def __init__(self):
+        self.meta = self.Meta(self.BilledUnits(9, 12))
+
+
+class CohereV2Response:
+    class BilledUnits:
+        def __init__(self, input_tokens: int, output_tokens: int):
+            self.input_tokens = input_tokens
+            self.output_tokens = output_tokens
+
+    class Usage:
+        def __init__(self, billed_units):
+            self.billed_units = billed_units
+
+    def __init__(self):
+        self.usage = self.Usage(self.BilledUnits(9, 12))
+
+
+class VertexAiResponse:
+    class UsageMetadata:
+        def __init__(self, promptTokenCount: int, candidatesTokenCount: int):
+            self.promptTokenCount = promptTokenCount
+            self.candidatesTokenCount = candidatesTokenCount
+
+    def __init__(self, promptTokenCount: int, candidatesTokenCount: int):
+        self.usageMetadata = self.UsageMetadata(promptTokenCount, candidatesTokenCount)
+
+
+class TestProcessLLMReponse(unittest.TestCase):
+    def test_anthropic_response(self):
+        response = AnthropicResponse("message", "claude-3-5-sonnet-20241022")
+        input_tokens_event, output_tokens_event = process_llm_response(
+            llm_response=response,
+            customer_id=customer_id,
+            aflo_dimensions=EMPTY_DIMENSIONS,
+        )
+
+        expected_dimensions = {
+            "model": response.model,
+            "object": response.type,
+            "provider": LlmProvider.ANTHROPIC.value,
+        }
+
+        self.assertIsNotNone(input_tokens_event[meter_value_key])
+        self.assertIsNotNone(input_tokens_event[unique_id_key])
+        self.assertIsNotNone(input_tokens_event[timestamp_key])
+        self.assertIsNotNone(input_tokens_event[customer_id_key])
+        self.assertIsNotNone(input_tokens_event[dimensions_key])
+        self.assertEqual(
+            input_tokens_event[meter_api_name_key], INPUT_TOKENS_METER_API_NAME
+        )
+        self.assertEqual(input_tokens_event[meter_value_key], 10)
+        self.assertEqual(input_tokens_event[dimensions_key], expected_dimensions)
+        self.assertEqual(input_tokens_event[customer_id_key], customer_id)
+
+        self.assertIsNotNone(output_tokens_event[meter_value_key])
+        self.assertIsNotNone(output_tokens_event[unique_id_key])
+        self.assertIsNotNone(output_tokens_event[timestamp_key])
+        self.assertIsNotNone(output_tokens_event[customer_id_key])
+        self.assertIsNotNone(output_tokens_event[dimensions_key])
+        self.assertEqual(
+            output_tokens_event[meter_api_name_key], OUTPUT_TOKENS_METER_API_NAME
+        )
+        self.assertEqual(output_tokens_event[meter_value_key], 12)
+        self.assertEqual(output_tokens_event[dimensions_key], expected_dimensions)
+        self.assertEqual(output_tokens_event[customer_id_key], customer_id)
```

Comment:
notice how these blocks of assertions repeat themselves. create a helper function to avoid duplication.